### PR TITLE
Avoid function constructor in d3

### PR DESCRIPTION
**Pull request by archmoj**
As mentioned in https://github.com/plotly/plotly.js/issues/897#issuecomment-744512711 `d3@3.5.17` (as well as [d3-dsv](https://www.npmjs.com/package/d3-dsv) use function constructors in the source.

This PR suggests refactoring d3 to avoid using function constructors.

[Demo](https://codepen.io/MojtabaSamimi/pen/mdrBOjr)

@plotly/plotly_js

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
     "color-rgba": "2.1.1",
     "convex-hull": "^1.0.3",
     "country-regex": "^1.1.0",
-    "d3": "^3.5.17",
+    "d3": "git://github.com/archmoj/d3.git#03a7fc0d448a3f58fa83ff08e6550c6659088fdb",
```
Comment (archmoj) [inline package.json:73]: See https://github.com/archmoj/d3/commit/93e74e544e9332ecbd17b43b3694de511310be6e

Comment (archmoj) [reply package.json:73]: Also see https://github.com/d3/d3-dsv/pull/67#issuecomment-749804075

**Review by alexcjohnson — APPROVED** [review]
💃 Nice - as mentioned on slack I'm comfortable with us using a fork of `d3v3` in the interim until we manage to upgrade to `d3v5`.

**Comment by nicolaskruchten** [comment]
This makes sense to me, and I still think we should stop exporting these functions and do a v2.

**Comment by nicolaskruchten** [comment]
Upon further reflection, although what I said above holds, there's no particular reason to do the v2 split immediately. I had initially thought that we would just not include the parts of d3v3 that use the function constructor, but if we're still exporting a fully-functional (although not up to Mike Bostock's performance standards!) version of d3v3 then we can just get on with things in a minor as usual.

**Pull request merged**